src/components/libvirtSlate.jsx

```diff
@@ -60,10 +61,19 @@ class LibvirtSlate extends React.Component {
     }
 
     startService() {
+        this.setState({ actionInProgress: true });
         const service = this.props.libvirtService;
 
-        enableLibvirt({ enable: this.state.libvirtEnabled, serviceName: service.name });
-        startLibvirt({ serviceName: service.name });
+        enableLibvirt({
+            enable: this.state.libvirtEnabled,
+            serviceName: service.name
+        })
+                .then(() => {
+                    startLibvirt({
+                        serviceName: service.name
+                    }).catch(ex => this.setState({ error: _("Failed to start virtualization service"), errorDetail: ex.message }));
+                }, ex => this.setState({ error: _("Failed to enable virtualization service"), errorDetail: ex.message }))
```

Comment:
It might be a bit simpler to read by chaining the two promises (i.e. drop the `{}` block around startLibvirt) and using a single exception handler for both. But this is not a strong opinion, if you like this better, by all means keep it.

Reply:
Right, I mainly did it to have the titles in errors Failed to start / Failed to enable. No strong opinion here either.